Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -341,17 +344,22 @@ def enableOrDisableCreateButton(self):
 
     def onInputCurveModified(self, observer, eventid):
         """If the input curve  was changed we need to repopulate the keyframe UI"""
-        pass
+        if not self.ignoreInputCurveModified:
+            if self.logic:
+                # We are going to have rebuild the EndoscopyLogic later
+                self.logic.cleanup()
+                self.logic = None
```

Comment:
Re-creating the logic upon modification should be avoided and is an anti-pattern, I have a follow-up change address this.

Reply:
Yes, we can re-architect it later -- good idea! -- but the current preview build does not work unless we do the full rebuild.

Reply:
> but the current preview build does not work unless we do the full rebuild.

Changes could be tested without rebuilding, you could for example download the latest Stable or Preview build and replace the `Endoscopy.py` file available in the install tree.

Alternatively, you could also remove the `Endoscopy.py` from the install tree and then drag and drop the one in your local checkout. See https://slicer.readthedocs.io/en/latest/developer_guide/extensions.html#run-slicer-with-your-custom-modules

Reply:
> preview build is broken if the user tries to modify the curve after having selected it in the Endoscopy module.

Thanks for clarifying. I will test an follow up.